rules-reviewed/technology-usage/tests/database-technology-usage.windup.test.xml

```diff
@@ -0,0 +1,79 @@
+    <testDataPath>data/database/</testDataPath>
+    <rulePath>../database-technology-usage.windup.xml</rulePath>
+    <ruleset>
+        <rules>
+            <rule id="technology-usage-database-01000-test">
+                <when>
+                    <not>
+                        <technology-statistics-exists name="JDBC datasources" number-found="1">
+                            <tag name="Store"/>
+                            <tag name="Database"/>
+                            <tag name="Java EE"/>
+                        </technology-statistics-exists>
+                    </not>
+                </when>
+                <perform>
+                    <fail message="JDBC datasources Technology Statistic Not Found" />
+                </perform>
+            </rule>
+            <rule id="technology-usage-database-01001-test">
+                <when>
+                    <not>
+                        <technology-statistics-exists name="JDBC XA datasources" number-found="1">
+                            <tag name="Store"/>
+                            <tag name="Database"/>
+                            <tag name="Java EE"/>
+                        </technology-statistics-exists>
+                    </not>
+                </when>
+                <perform>
+                    <fail message="JDBC datasources XA Technology Statistic Not Found" />
+                </perform>
+            </rule>
+            <rule id="technology-usage-database-01100-test">
+                <when>
+                    <not>
+                        <technology-statistics-exists name="JPA entities" number-found="1">
+                            <tag name="Store"/>
+                            <tag name="Persistence"/>
+                            <tag name="Java EE"/>
+                        </technology-statistics-exists>
+                    </not>
+                </when>
+                <perform>
+                    <fail message="JPA entities Technology Statistic Not Found" />
+                </perform>
+            </rule>
+            <rule id="technology-usage-database-01200-test">
+                <when>
+                    <not>
+                        <technology-statistics-exists name="JPA named queries" number-found="1">
+                            <tag name="Store"/>
+                            <tag name="Persistence"/>
+                            <tag name="Java EE"/>
+                        </technology-statistics-exists>
+                    </not>
+                </when>
+                <perform>
+                    <fail message="JPA named queries Technology Statistic Not Found" />
+                </perform>
+            </rule>
+            <rule id="technology-usage-database-01300-test">
+                <when>
+                    <not>
+                        <technology-statistics-exists name="Persistence units" number-found="1">
```

Comment:
For this test to work, it's has to be added in the `data/database/` folder at least one file with a persistence unit like [persistence.xml](https://github.com/windup/windup-rulesets/blob/master/rules-reviewed/technology-usage/tests/data/jta/persistence.xml).
With such a file also test `technology-usage-database-01000-test` and `technology-usage-database-01001-test` succeed then.

Reply:
I added `persistence.xml` file. But tests `0100` and `01001` should not be related to that. There is `mysql-ds.xml` file which defines data sources. Or it needs to be in `persistence.xml` file?

Reply:
I didn't find any Java rule for creating a `DataSourceModel` from a  `*-ds.xml` file.
`*-ds.xml` files are only handled by rules in [`eap4-xml-config.windup.xml`](https://github.com/windup/windup-rulesets/blob/master/rules-reviewed/eap6/jboss-eap4/eap4-xml-config.windup.xml#L31) file but those rules creates only hints right now.
You can add in  `eap4-eap6-25000`, `eap4-eap6-26000` and `eap4-eap6-27000` rules the tag
```xml
<technology-tag level="INFORMATIONAL">Datasource</technology-tag>
```
and then, in `technology-usage-database-01000` rule, replace the current `<when>` block with:
```xml
            <when>
                <graph-query discriminator="TechnologyTagModel">
                    <property name="name">Datasource</property>
                </graph-query>
            </when>
```

Reply:
It seems like we really identify datasources only in `persistence.xml` file by using  `DiscoverJpaConfigurationXmlRuleProvider`.

I would rather create some special rule for identifying datasources, then modifying those you mentioned. Because the won't be executed in case of selecting any other target, then EAP 6, right?